src/popup.c

```diff
@@ -58,6 +58,27 @@ void ShutdownPopup(void)
    }
 }
 
+/** Calculate dimensions of a popup window given the popup text. */
+char** MeasurePopupText(const char *text, int *width, int *height, int *rows) {
+    char **result = NULL;
+    *width  = 0;
+    *height = 1;
+    *rows   = 0;
+
+    for (char *ptr = strtok(strdup(text), "\n"); ptr != NULL; ptr = strtok(NULL,"\n"))
```

Comment:
I don't think Joe requires c99... probably should move char *ptr to function header.